spec/helpers.lua

```diff
@@ -892,6 +892,33 @@ local function wait_for_invalidation(key, timeout)
   end, timeout)
 end
 
+--- Waits for the termination of a pid.
+-- @param pid_path Filename of the pid file.
+-- @param timeout (optional) in seconds, defaults to 10.
+local function wait_pid(pid_path, timeout)
+  local pid
+  local fd = io.open(pid_path)
+  if fd then
+    pid = fd:read("*l")
+    fd:close()
+  end
+
+  if pid then
+    local max_time = ngx.now() + (timeout or 10)
+    repeat
+      if pl_utils.execute("ps -p " .. pid .. " >/dev/null 2>&1") then
+        ngx.sleep(0.05)
+      else
+        return
+      end
+    until ngx.now() >= max_time
+    -- Timeout: kill with SIGKILL
+    pl_utils.execute("kill -9 " .. pid .. " >/dev/null 2>&1")
+    -- Sanity check: check pid again, but don't loop.
+    wait_pid(pid_path, timeout, false)
```

Comment:
It doesn't seem like `wait_pid`'s signature accepts a third argument?

Reply:
Good catch. I toyed with the idea of making `kill -9` optional, but decided it would be an unnecessary moving part.

Reply:
I think this one would be the only blocker at this point. Other changes can be dealt with. In the future, using this in `kong stop` will also help making the CLI tests (that rely on `kong stop` and `kong start`) more reliable as well.